content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx

```diff
@@ -0,0 +1,159 @@
+For example, the following code uses a [`model` extension](/concepts/components/prisma-client/client-extensions/model) to extend the `user` model.
+
+```ts
+const xprisma = prisma.$extends({
+  name: `signUp`,  // Optional name appears in error logs and tracing
+  model: {  // This is a `model` type extension
+    user: { ... } // The extension logic goes inside the curly braces
+ },
+})
+```
+
+`name` is an optional field that you can use to name the extension. This name appears in error logs and tracing.
+
+## Multiple extensions
+
+When you create multiple extensions, there are two ways to associate them with extended clients:
+
+- Create each extension in a separate extended client.
+- Combine two or more extensions into one extended client. The functionality from these extensions applies in the same client instance.
+
+  Note: [Combined extensions can conflict](#conflicts-in-combined-extensions).
+
+You can combine the two approaches above. For example, you could have one extension in its own extended client and combine two other extensions into another extended client.
+
+[Learn more about how client instances interact](#client-instances)
+
+### Apply multiple extensions to one or more extended clients
+
+In the following example, suppose that you have two extensions, `extensionA` and `extensionB`. There are two ways to combine these into a client.
+
+First, declare the standard Prisma client as usual:
+
+```ts
+// First of all, store your original Prisma Client in a variable as usual
+const prisma = new PrismaClient()
+```
+
+** Option 1: declare the new client in one line **
+
+With this option, you apply both extensions to a new client in one line of code.
+
+```ts
+// Declare an extended client that has extensionsA and extensionB
+const clientAB = prisma.$extends(extensionA).$extends(extensionB)
+```
+
+You can refer to `clientAB` in your code, for example `clientAB.getName()`.
+
+** Option 2: declare multiple extended clients **
+
+The advantage of this option is that you can call any of the extended clients separately.
+
+```ts
+// Declare an extended client that has extension A applied
+const clientA = prisma.$extends(extensionA)
+
+// Declare an extended client that has extension B applied
+const clientB = prisma.$extends(extensionB)
+
+// Declare an extended client that is a combination of clientA and extensionB
+const clientAplusB = clientA.$extends(extensionB)
+```
+
+In your code, you can call any of these clients separately, for example `clientA.getName()`, `clientB.getName()`, or `clientAplusB.getName()`.
+
+### Conflicts in combined extensions
+
+When you combine two or more extensions, then the _last_ extension that you declare takes precedence in any conflict. In the example in option 1 above, suppose there is a method called `getName()` in `clientA` and a method called `getName()` in `clientB`. When you call `clientAB.getName()`, then the client uses `getName()` as defined in `clientB`.
+
+## Extensions versus middlewares
```

Comment:
I am not sure whether this belongs here. My thinking is that it is the `query` extension layer that is a replacement for middlewares, here it is comparing the whole concept of extensions to middlewares, which seems incorrect since middlewares cannot do `model`, `client`, or `result` extensions.

Reply:
Ah, OK, I didn't realize that. I fully agree! I'll do the following:
1. Move this section to the `query` page. 
2. Add a note to the summary of each extension type (the bulleted list) in the intro